packages/x-date-pickers/src/internals/hooks/useField/useFieldRootHandleKeyDown.ts

```diff
@@ -195,8 +195,7 @@ function adjustSectionValue<TValue extends PickerValidValue>(
     const step =
       section.type === 'minutes' && stepsAttributes?.minutesStep ? stepsAttributes.minutesStep : 1;
 
-    const currentSectionValue = parseInt(removeLocalizedDigits(section.value, localizedDigits), 10);
-    let newSectionValueNumber = currentSectionValue + delta * step;
+    let newSectionValueNumber: number;
```

Comment:
Not related, it was just weird to generate `NaN` on `newSectionValueNumber` and then not use it, I lost some time debugging because of it